src/components/lib/FlowGrid/FlowGrid.jsx

```diff
@@ -198,19 +220,40 @@ export default class FlowGrid extends Component{
     this._addIfNeededAndSelect(newLocation, isRight ? 'LEFT' : 'RIGHT')
   }
 
+  onFillTargetMouseDown(location) {
+    window.recorder.recordNamedEvent("FlowGrid set fillRegion state")
+    this.setState({tracingFillRegion: true, fillRegion: {start: location}})
+  }
+
+  onCellMouseUp(location) {
+    if (!this.state.tracingFillRegion) { return }
+    window.recorder.recordNamedEvent("FlowGrid set fillRegion state")
+    this.props.onFillRegion(this.state.fillRegion)
+    this._handleEndRangeSelect(this.state.fillRegion.end)
+    this.setState({tracingFillRegion: false, fillRegion: {}})
+  }
+
+  // TODO(matthew): Look into necessity of 'inSelectedRegion' passed to cell below.
```

Comment:
I want to look into it later, so I'm leaving the TODO there for future reference.